cmd/manager/cleanup.go

```diff
@@ -0,0 +1,178 @@
+	options.SetLabelSelector(fmt.Sprintf("csc-owner-name"))
+
+	subscriptions := &olm.SubscriptionList{}
+	// Get the list of existing subscriptions that have the label "csc-owner-name"
+	err := kubeClient.List(context.TODO(), options, subscriptions)
+	if err != nil {
+		return nil, err
+	}
+	for _, instance := range subscriptions.Items {
+		cscNames = append(cscNames, instance.GetLabels()["csc-owner-name"])
+		// try to guess the datastore catalogsource from the subscription
+		datastoreCs, err := findCatalogSource(&instance, kubeClient)
+		if k8s_errors.IsNotFound(err) {
+			// guess the catalogsource from the opsrc that has the package
+			datastoreCs, err = findCsFromOpsrc(&instance, kubeClient)
+			if err != nil {
+				return nil, err
+			}
+		}
+		// point the subscription to the datastore CatalogSource
+		instance.SetCatalogSource(datastoreCs.GetName())
+		instance.SetCatalogSourceNamespace("openshift-marketplace")
+		kubeClient.Update(context.TODO(), &instance)
+	}
+	return cscNames, nil
+}
+
+func deleteDatastoreCscs(cscNames []string, kubeClient client.Client) error {
+	for _, cscName := range cscNames {
+		namespacedName := types.NamespacedName{Name: cscName, Namespace: "openshift-marketplace"}
+		csc := &marketplace.CatalogSourceConfig{}
+		err := kubeClient.Get(context.TODO(), namespacedName, csc)
+		if err == nil {
+			err = kubeClient.Delete(context.TODO(), csc)
+			if err != nil {
+				return err
+			}
+		} else if err != nil && !k8s_errors.IsNotFound(err) {
+			return err
+		}
+
+	}
+	return nil
+}
+
+func deleteOldCscs(kubeClient client.Client) error {
+	options := &client.ListOptions{}
+	options.SetLabelSelector(fmt.Sprintf("opsrc-datastore: \"true\""))
+
+	cscs := &marketplace.CatalogSourceConfigList{}
+	// Get the list of existing cscs that have the lable "opsrc-datastore: "true" "
+	err := kubeClient.List(context.TODO(), options, cscs)
+	if err != nil {
+		return err
+	}
+	for _, csc := range cscs.Items {
+		kubeClient.Delete(context.TODO(), &csc)
+	}
+	return nil
+}
+
+func findCatalogSource(subscription *olm.Subscription, client client.Client) (*olm.CatalogSource, error) {
+	associatedCscName := subscription.GetLabels()["csc-owner-name"]
+	possibleCsName, err := extractCsName(associatedCscName)
+	if err != nil {
+		return nil, err
+	}
+	// try and fetch the CatalogSource
+	datastoreCs := &olm.CatalogSource{}
+	namespacedName := types.NamespacedName{Name: possibleCsName, Namespace: "openshift-marketplace"}
+	err = client.Get(context.TODO(), namespacedName, datastoreCs)
+	if err != nil {
+		return nil, err
+	}
+	return datastoreCs, nil
+}
+
+func findCsFromOpsrc(subscription *olm.Subscription, clt client.Client) (*olm.CatalogSource, error) {
+	packageName := subscription.GetName()
+	opsrcs := &marketplace.OperatorSourceList{}
+	err := clt.List(context.TODO(), &client.ListOptions{}, opsrcs)
+	if err != nil {
+		return nil, err
+	}
+	for _, instance := range opsrcs.Items {
+		if isPackageInOpsrc(packageName, &instance) {
```

Comment:
Code is easier to read if you check the negation here. 
```
if !isPackageInOpsrc(packageName, &instance) {
    continue
}
```
This will result in less tabbed code below.